content/posts/how-to-use-playwright-in-cucumberjs/index.md

```diff
@@ -0,0 +1,157 @@
+- Intercept network activity for stubbing and mocking network requests
+- Emulate mobile devices, geolocation, permissions
+- Support for web components via shadow-piercing selectors
+- Native input events for mouse and keyboard
+- Upload and download files
+
+### Getting Started with Cucumber and Playwright Example
+
+#### Prerequisites and Installations
+- Prerequisites:
+  - [Node.js](https://nodejs.org/en/) (12 or higher)
+- Installations:
+  - Install Cucumber modules with [yarn](https://yarnpkg.com/en/) or [npm](https://www.npmjs.com/)
+    - yarn:
+
+        ```bash
+        yarn add @cucumber/cucumber
+        ```
+    - npm:
+
+        ```bash
+        npm i @cucumber/cucumber
+        ```
+
+  - Install Playwright
+    - yarn:
+        ```bash
+        yarn add playwright
+        ```
+    - npm:
+        ```bash
+        npm i playwright
+        ```
+    - Add the following files:
+      - `features/search_job_openings_at_manabie.feature`
+
+        ```feature
+        Feature: Search job openings at Manabie
+
+            Scenario: Bob search job openings at Manabie
+                Given Bob opens Manabie website
+                When Bob goes to Careers section
+                Then Bob sees all job openings at Manabie
+        ```
+      - `features/support/world.js`
+
+        ```javascript
+        const { setWorldConstructor } = require("@cucumber/cucumber");
+        const playwright = require('playwright');
+
+        class CustomWorld {
+            async openUrl(url) {
+                const browser = await playwright.chromium.launch({
+                    headless: false,
+                });
+                const context = await browser.newContext();
+                this.page = await context.newPage();
+                await this.page.goto(url);
+            }
+        }
+
+        setWorldConstructor(CustomWorld);
+        ```
+
+      - `features/support/steps.js`
+
+        ```javascript
+        const { Given, When, Then } = require("@cucumber/cucumber");
+
+        Given("Bob opens Manabie website", { timeout: 60 * 1000 }, async function () {
+            await this.openUrl('http://manabie.com/');
+        });
+
+        When("Bob goes to Careers section", async function () {
+            await this.page.click('text=Careers');
+        });
+
+        Then("Bob sees all job openings at Manabie", async function () {
+            await this.page.click('text=View Openings');
+            await this.page.waitForSelector('text=Our Openings');
+        });
+        ```
+
+    - Run:
+
+        ```bash
+        ./node_modules/.bin/cucumber-js --exit
+        ```
+
+    - After run:
+
+        ```bash
+        1 scenario (1 passed)
+        3 steps (3 passed)
+        0m03.739s (executing steps: 0m03.729s)
+        ```
+
+### Conclusion
+Cucumber and Playwright is great framework. I hope this article will be of some use to you. [Here is source code](/content/posts/how-to-use-playwright-in-cucumberjs/example), thank you.
```

Comment:
Cucumber and Playwright are great frameworks. I hope this article will be of some use to you. [Here is source code](/content/posts/how-to-use-playwright-in-cucumberjs/example), thank you.

Curious for more about how we're tackling testing problems at Manabie? Join us as an Engineer **[HERE](https://manabie.breezy.hr/)**, or follow our blog for future articles.